lib/log.h

```diff
@@ -77,10 +77,17 @@ extern void metal_default_log_handler(enum metal_log_level level,
  * @param	level	Log level.
  * @param	...	Format string and arguments.
  */
-#define metal_log(level, ...)						       \
-	((level <= _metal.common.log_level && _metal.common.log_handler) \
-	       ? (void)_metal.common.log_handler(level, __VA_ARGS__)	       \
-	       : (void)0)
+#define metal_log(level, fmt, args ...)					\
+do {									\
+	if (_metal.common.log_handler && level <= _metal.common.log_level) \
+		_metal.common.log_handler(level, "%s:%u " fmt "\n",	\
+					  __func__, __LINE__, ##args);	\
+} while (0)
```

Comment:
So, I checked size of libmetal before and after this patch with this command: "size libmetal.a"

I see before .text section: 25187 i.e. ~25.1K
After .text section: 23958 i.e. ~23.9K. 
~1.2k difference.

Macros are certainly adding values. But, __func__, __LINE__ and "\n" forcing in all the logs may not be appreciated by every user.
I prefer we leave that choice to user.

Reply:
Agree with @tnmysh 
 One of the pillars of the library is to be able to run with a minimal footprint. If the addition of functions name and lines should impact the nominal footprint this should be something that is disable by default.